.github/workflows/release-check.yml

```diff
@@ -2,134 +2,165 @@ name: Release Checker
 on: [ workflow_call ]
 
 jobs:
-  releaser:
+  prepare:
     runs-on: ubuntu-latest
     env:
-      TAG_EXISTS: "true"
-      VERSION: "" # the version number read from version.json
-      COMPARETO: "" # the version number to compare this version to
-      GORELEASE: ""
-      GOCOMPAT: ""
-      GOMODDIFF: ""
-      RELEASE_BRANCH_NOTE: ""
+      GITHUB_TOKEN: ${{ github.token }}
+    outputs:
+      version: ${{ steps.version.outputs.this }}
+      tag_exists: ${{ steps.tag_exists.outputs.this }}
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-go@v3
-        with:
-          go-version: "1.19.x"
-      - name: Determine version
-        env:
-          GITHUB_TOKEN: ${{ github.token }}
+      - id: version
+        name: Retrieve new version from version.json
         run: |
-          # If `version.json` file doesn't exists, `version` is `""` and `404` is printed on stderr.
-          # The step won't be marked as a failure though because the error happens in a subshell.
-          version="$(gh api -X GET 'repos/${{ github.event.pull_request.head.repo.full_name }}/contents/version.json' -f ref='${{ github.event.pull_request.head.sha }}' --jq '.content' | base64 -d | jq -r '.version')"
+          # https://docs.github.com/en/rest/reference/repos#get-repository-content
+          version="$(gh api -X GET 'repos/${{ github.event.pull_request.head.repo.full_name }}/contents/version.json' -f ref='${{ github.event.pull_request.head.sha }}' --jq '.content' | base64 -d | jq -r '.version // ""')"
+          status=$? && ([[ $status == 0 ]] || exit $status)
           echo "version=$version"
-          echo "VERSION=$version" >> $GITHUB_ENV
-      - name: Check if the tag already exists
-        # Check if a git tag for the version (as read from version.json) exists
-        # If that is the case, we don't need to run the rest of the workflow.
-        if: env.VERSION != ''
+          echo "::set-output name=this::$version"
+      - id: tag_exists
+        name: Check if version tag already exists
+        if: steps.version.outputs.this != ''
         run: |
-          git fetch origin --tags
-          status=0
-          git rev-list $VERSION &> /dev/null || status=$?
-          if [[ $status != 0 ]]; then
-            echo "TAG_EXISTS=false" >> $GITHUB_ENV
-          fi
-      - name: Install semver (node command line tool)
-        if: env.TAG_EXISTS == 'false'
-        run: npm install -g "https://github.com/npm/node-semver#e79ac3a450e8bb504e78b8159e3efc7089569" # v7.3.5
+          # https://docs.github.com/en/rest/reference/git#get-a-reference
+          gh api '/repos/${{ github.repository }}/git/ref/tags/${{ steps.version.outputs.this }}' && tag_exists='true' || tag_exists='false'
```

Comment:
We're using string interpolation at the yml level here, is it possible a malformed value in `steps.version.outputs.this`  turns into a string injection?

Should we use something like:

```yml
env:
    TAG_URL: /repos/${{ github.repository }}/git/ref/tags/${{ steps.version.outputs.this }}
run:
    gh api "${TAG_URL}" # using shell's expansion here
```

?

ref: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable